File: Easy_Level_using_Cpp/Array and String/217. Contains Duplicate.cpp
// Problem: 217. Contains Duplicate
// Link: https://leetcode.com/problems/contains-duplicate/description/
// Time Complexity: O(n)
// Space Complexity: O(n)
// Topics: Array, Hash Table, Sorting
// Solution:
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        if(nums.size() == 1){
            return false;
        }
        unordered_set<int> seen; // Created Hash set

        for(int num: nums){      // Iterating through array
            if (seen.count(num)){  // Number already existing in hash?
                return true;    // Duplicate Found
            }
            seen.insert(num);   // insert the element in hash
        }

        return false; // All unique
    }
};

File: Easy_Level_using_Python/Array and String/217. Contains Duplicate.py
# Problem: 217. Contains Duplicate
# Link: https://leetcode.com/problems/contains-duplicate/description/
# Time Complexity: O(n)
# Space Complexity: O(n)
# Topics: Array, Hash Table, Sorting
# Solution:
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        if(len(nums) == 1):
            return False

        seen = set()

        for num in nums:
            if num in seen:
                return True
            seen.add(num)

        return False
        
        
        
        
        